FreeRTOS/FreeRTOS-Plus-TCP
# [IPv6] Check unspecified/loopback/multicast reserved/matching addresses in prvAllowIPPacketIPv6.

[IPv6] Check unspecified/loopback/multicast reserved/matching addresses in prvAllowIPPacketIPv6.

Description
-----------

From [RFC 4291](https://www.rfc-editor.org/rfc/rfc4291.html), some IPv6 addresses are reserved and shouldn't be used. Drop packets when the address met below conditions:
1. Source or destination address have unspecified address (All 0s).
2. Check endpoint's IP address with packet's destination.
3. Allow loopback packets from this node itself only.
4. Check if multicase address isn't valid. (Refer to [sec 2.7](https://www.rfc-editor.org/rfc/rfc4291.html#section-2.7) in RFC4291)
5. Fix coverity warnings.

Test Steps
-----------

Run protocol tests UDP/Others/003, UDP/Others/004, IPv6/datagram/010~017, 019.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
